### PR TITLE
Fix xml filename

### PR DESCRIPTION
**Pull request by tank157**
* Upgrade zip4j dependency
* Make the filename generation a bit more generic so we can use it for both XML and ZIP filenames
* Rename the temp xml file from the pseudo-random name that all temp files get by default to the well structured name

### CODE REVIEW

File: src/vip/data_processor/output/xml_helpers.clj [hunk 1]
```diff
@@ -1,12 +1,74 @@
...
+            [vip.data-processor.s3 :as s3]
+            [vip.data-processor.util :as util])
+  (:import [java.nio.file Files CopyOption StandardCopyOption]
            [java.nio.file.attribute FileAttribute]))
 
+(defn format-fips [fips]
+  (cond
+    (empty? fips) "XX"
+    (< (count fips) 3) (format "%02d" (Integer/parseInt fips))
+    (< (count fips) 5) (format "%05d" (Integer/parseInt fips))
```
Comment (alpha-dw) [inline src/vip/data_processor/output/xml_helpers.clj:14]: Contextual question: Is the check for less than 5 made to account for a previous version of FIPS codes ([is this the one?](https://en.wikipedia.org/wiki/FIPS_county_code)? I am wondering why it is not checking for less than 6, but perhaps that is due to my lack of context!

Comment (tank157) [reply src/vip/data_processor/output/xml_helpers.clj:14]: TBH I'm not sure, this was part of some code I simply moved to a new location. If I had to guess, though, I'd suspect it's to handle one of several ways in which FIPS codes for Alabama, Alaska, Arizona, Arkansas, California, Colorado and Connecticut counties, all of whom start with a leading 0. It's super easy for a human to not understand that the leading 0 is important, or for a number parser to strip it off. There's still some troubling gaps in there, as mostly that case I would expect to be exactly 4 digits, but I can't figure out a valid case where we'd only have 3 digits, aside from a county accidentally _only_ providing the county portion of their full FIPS.

Comment (alpha-dw) [reply src/vip/data_processor/output/xml_helpers.clj:14]: 🤔

Comment (tank157) [reply src/vip/data_processor/output/xml_helpers.clj:14]: This commit from the original place the code appears is interesting: https://github.com/votinginfoproject/data-processor/commit/a6a10e3abe21b5ab75bc00a298d42cdea7dd39e4

So it definitely started out as a "zero-padding" thing from what I can tell.

Comment (alpha-dw) [reply src/vip/data_processor/output/xml_helpers.clj:14]: Oh, that's an interesting fix. I wonder what the case was...

Comment (tank157) [reply src/vip/data_processor/output/xml_helpers.clj:14]: I think the comment on the commit helps clarify. When the first comparison was "less than 2", all 2 digit FIPS codes were getting 0-padded to 5 digits, but all state FIPS codes are actually 2 digits so they were getting needlessly padded.

Comment (alpha-dw) [reply src/vip/data_processor/output/xml_helpers.clj:14]: Ah yes, missed that. That makes sense. And given the context you provided regarding the various FIPS, it looks clear enough for me

File: src/vip/data_processor/output/xml_helpers.clj [hunk 2]
```diff
@@ -1,12 +1,74 @@
...
+             (filename* fips state election-date)))))
+
 (defn create-xml-file
-  [{:keys [import-id] :as ctx}]
-  (let [xml-file (Files/createTempFile
-                  (str import-id) ".xml" (into-array FileAttribute []))]
+  "Creates a temp file for the output xml, then moves it to a name without
+   the randomness that temp files get automatically added to their names."
+  [{:keys [import-id output-file-basename] :as ctx}]
+  (let [tmp-xml-file (Files/createTempFile
```
Comment (lizstarin) [inline src/vip/data_processor/output/xml_helpers.clj:65]: Java doesn't let you create a temp file where you just supply the full name, huh?

File: test/vip/data_processor/output/xml_helpers_test.clj [hunk 3]
```diff
@@ -1,9 +1,38 @@
...
+
+  (testing "workably formatted dates make nice filenames"
+    (is (= "vipfeed-52-FL-2015-03-27"
+           (filename* "52" "FL" "2015/03/27"))))
+
+  (testing "poorly formatted dates use a workable filename"
+    (is (= "vipfeed-12345-North-Carolina-"
+           (filename* "12345" "North Carolina" nil))))
+
+  (testing "missing the fips and/or state doesn't break the world"
```
Comment (lizstarin) [inline test/vip/data_processor/output/xml_helpers_test.clj:19]: 😆

File: src/vip/data_processor/output/xml_helpers.clj (hunk 2 above)
Comment (tank157) [reply src/vip/data_processor/output/xml_helpers.clj:65]: Sadly no. I suspect it's to allow you to create many temp files with the same code over and over again, but it's annoying for sure.

File: src/vip/data_processor/output/xml_helpers.clj [hunk 4]
```diff
@@ -1,12 +1,74 @@
...
-  (:import [java.nio.file Files]
+  (:require [clojure.string :as str]
+            [korma.core :as korma]
+            [vip.data-processor.db.postgres :as postgres]
+            [vip.data-processor.s3 :as s3]
+            [vip.data-processor.util :as util])
+  (:import [java.nio.file Files CopyOption StandardCopyOption]
            [java.nio.file.attribute FileAttribute]))
 
+(defn format-fips [fips]
```
Comment (lizstarin) [inline src/vip/data_processor/output/xml_helpers.clj:10]: Were there tests for these formatting functions, previously?

Comment (tank157) [reply src/vip/data_processor/output/xml_helpers.clj:10]: Not that I saw. Happy to add one though

Comment (tank157) [reply src/vip/data_processor/output/xml_helpers.clj:10]: Added some tests for the formatting functions.

**Pull request merged**